convert2rhel/pkghandler.py

```diff
@@ -1079,52 +1054,170 @@ def _get_packages_to_update_dnf(reposdir):
-    :type pkg_2: str
-    :return: An integer resulting in the package comparision
+        Since the return type is a int, this could be difficult to understand
+        the meaning of each number, so here is a list that represents
+        every possible number:
+            * -1 if the version1 is less then version2 version
+            * 0 if the version1 is equal version2 version
+            * 1 if the version1 is greater than version2 version
+
+    :return: Return a number indicating if the versions match, are less or greater then.
     :rtype: int
     """
+    # call parse_pkg_string to obtain a list containing name, epoch, release, version and arch fields
+    # order of fields returned: name, epoch, version, release, arch
+    version1_components = parse_pkg_string(version1)
+    version2_components = parse_pkg_string(version2)
+
+    # ensure package names match, error if not
+    if version1_components[0] != version2_components[0]:
+        raise ValueError("%s does not match %s" % (version1_components[0], version2_components[0]))
+
+    # ensure package arches match, error if not
+    if version1_components[4] != version2_components[4] and all(([version1_components[4]], version2_components[4])):
+        raise ValueError("%s does not match %s" % (version1_components[4], version2_components[4]))
+
+    # create list containing EVR for comparison
+    evr1 = (version1_components[1], version1_components[2], version1_components[3])
+    evr2 = (version2_components[1], version2_components[2], version2_components[3])
+
+    return rpm.labelCompare(evr1, evr2)
+
+
+def parse_pkg_string(pkg):
+    """
+    This function takes a version string in NEVRA, NEVR, NVRA, NVR, ENVRA, ENVR and decides whether
+    to parse with a yum/dnf module based on the package manager type of the system.
+
+    :param pkg: The package to be parsed.
+    :type pkg: str
 
-    # TODO(r0x0d): This function still needs some enhancements code-wise, it
-    # workes perfectly, but the way we are handling the versions is not 100%
-    # complete yet.  will be done in a future work. Right now, all the list of
-    # changes are listed in this comment:
-    # https://github.com/oamg/convert2rhel/pull/469#discussion_r873971400
-    pkg_ver_components = []
-    for pkg in pkg_1, pkg_2:
-        # Remove the package name and split the rest between epoch + version
-        # and release + arch
-        epoch_version, release_arch = pkg.rsplit("-", 2)[-2:]
-        # Separate the (optional) epoch from the version
-        epoch_version = epoch_version.split(":", 1)
-        if len(epoch_version) == 1:
-            epoch = "0"
-            version = epoch_version[0]
+    :return: Return a Return a list containing name, epoch, version, release, arch
+    :rtype: list[str | None]
+    """
+    if pkgmanager.TYPE == "yum":
+        pkg_ver_components = _parse_pkg_with_yum(pkg)
+    else:
+        pkg_ver_components = _parse_pkg_with_dnf(pkg)
+
+    return pkg_ver_components
```

Comment:
Some of this should be moved into `pkgmanager/handlers/{yum.dnf}`.  @r0x0d was going to look into what makes sense to do there going forward, though, so t can be done at a later time.  I just wanted to note it so that it can be thought about when you design functions that have separate backends for yum and dnf in the future.

Reply:
Noted thank you

Reply:
Well noted, @abadger. Maybe I can include this comment in the jira issue where we are tracking this progress. I know that I thought about that the other day, but probably forgot to update comments/jira.